tendrl/ceph_integration/objects/rbd/atoms/create/__init__.py

```diff
@@ -13,6 +17,54 @@ def __init__(self, *args, **kwargs):
         super(Create, self).__init__(*args, **kwargs)
 
     def run(self):
+        # Checking if a new  pool has to be created or not rbd creation.
+        if self.parameters['Rbd.create_pool'] is True and \
+                        self.parameters['Rbd.pool_id'] is None:
+            # Creating job for pool creation.
+            if ['Rbd.pool_parameters'] in self.parameters:
```

Comment:
Id suggest we re-use (add to Rbd obj definitions) these attributes in the Rbd object, this maintains a relation between Rbd and its backing pool.
so redefine these attributes in the Rbd object (
"pool_poolname"
"pool_pg_num"
"pool_min_size"
"pool_size"
"pool_id"
}

in cases where the pool_id is provided by the user during the Rbd create call, you store Rbd.pool_id and skip rest of the above attributes.